client.go

```diff
@@ -754,6 +754,10 @@ func (c *client) SubscribeMultiple(filters map[string]byte, callback MessageHand
 			return token
 		}
 	}
+	if len(filters) == 0 {
+		token.setError(fmt.Errorf("topic filter/qos map must not be empty"))
+		return token
+	}
 	sub := packets.NewControlPacket(packets.Subscribe).(*packets.SubscribePacket)
 	if sub.Topics, sub.Qoss, err = validateSubscribeMap(filters); err != nil {
```

Comment:
One idea I was unsure about was to add this error check to `validateSubscribeMap` because it might make more sense to group the topic/qos validation logic together. On the other hand, this is a simple check that could be performed as early as possible. Please let me know if you have a preference as to where this check should be performed and I'd be happy to make the adjustment.

Reply:
If you could move this into the validateSubscribeMap option that would be great, subscribes aren't usually a frequent operation so moving this effectively a couple of lines further down should be fine and keeps the flow of the original